File: release_check.py
# SPDX-License-Identifier: MIT
"""Validate public desktop runtime metadata before packaging a release."""
from __future__ import annotations

from runtime import verify_release_sources


def main() -> int:
    """Print every verified default transport and exit unsuccessfully on drift."""
    for source in verify_release_sources():
        print(f"verified: {source}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: crates/dsh-core/examples/release_check.rs
fn main() {
    match dsh_core::runtime::verify_release_sources() {
        Ok(sources) => {
            for source in sources {
                println!("verified: {source}");
            }
        }
        Err(error) => {
            eprintln!("release source verification failed: {error}");
            std::process::exit(1);
        }
    }
}
